src/dash-table/components/EdgeFactory.tsx

```diff
@@ -0,0 +1,314 @@
+import { EdgesMatrices, IEdgesMatrices } from 'dash-table/derived/edges/type';
+
+import getHeaderRows from 'dash-table/derived/header/headerRows';
+
+import { derivedRelevantCellStyles, derivedRelevantFilterStyles, derivedRelevantHeaderStyles } from 'dash-table/derived/style';
+import { Style, Cells, DataCells, BasicFilters, Headers } from 'dash-table/derived/style/props';
+
+import { ControlledTableProps, VisibleColumns, IViewportOffset, Data, ICellCoordinates } from './Table/props';
+import { SingleColumnSyntaxTree } from 'dash-table/syntax-tree';
+
+type EdgesMatricesOp = EdgesMatrices | undefined;
+
+export default class EdgeFactory {
+    private readonly dataStyles = derivedRelevantCellStyles();
+    private readonly filterStyles = derivedRelevantFilterStyles();
+    private readonly headerStyles = derivedRelevantHeaderStyles();
+
+    private readonly getDataEdges = derivedDataEdges();
+    private readonly getDataOpEdges = derivedDataOpEdges();
+    private readonly getFilterEdges = derivedFilterEdges();
+    private readonly getFilterOpEdges = derivedFilterOpEdges();
+    private readonly getHeaderEdges = derivedHeaderEdges();
+    private readonly getHeaderOpEdges = derivedHeaderOpEdges();
+
+    private static clone(target: EdgesMatricesOp) {
+        return target && target.clone();
+    }
+
+    private static hasPrecedence(target: number, other: number, cutoff: number): boolean {
+        return (other <= cutoff || target === Infinity) && other <= target;
+    }
+
+    private hOverride(previous: EdgesMatricesOp, target: EdgesMatricesOp, cutoffWeight: number) {
+        if (!previous || !target) {
+            return;
+        }
+
+        const hPrevious = previous.getMatrices().horizontal;
+        const hTarget = target.getMatrices().horizontal;
+
+        const iPrevious = hPrevious.rows - 1;
+        const iTarget = 0;
+
+        R.forEach(j =>
+            EdgeFactory.hasPrecedence(
+                hPrevious.getWeight(iPrevious, j),
+                hTarget.getWeight(iTarget, j),
+                cutoffWeight
+            ) && hTarget.setEdge(iTarget, j, hPrevious.getEdge(iPrevious, j), Infinity, true)
+            , R.range(0, hPrevious.columns)
+        );
+    }
+
+    private vOverride(previous: EdgesMatricesOp, target: EdgesMatricesOp, cutoffWeight: number) {
+        if (!previous || !target) {
+            return;
+        }
+
+        const hPrevious = previous.getMatrices().vertical;
+        const hTarget = target.getMatrices().vertical;
+
+        const jPrevious = hPrevious.columns - 1;
+        const jTarget = 0;
+
+        R.forEach(i =>
+            EdgeFactory.hasPrecedence(
+                hPrevious.getWeight(i, jPrevious),
+                hTarget.getWeight(i, jTarget),
+                cutoffWeight
+            ) && hTarget.setEdge(i, jTarget, hPrevious.getEdge(i, jPrevious), Infinity, true)
+            , R.range(0, hPrevious.rows)
+        );
+    }
```

Comment:
Evaluates boundary edges between the previous and the target -- and overrides the target's value with the previous value if it does not have precedence. This handles the fixed columns/rows cases